packages/ui5-tooling-modules/lib/utils/WebComponentRegistry.js

```diff
@@ -390,6 +403,45 @@ class RegistryEntry {
 		}
 	}
 
+	/**
+	 * Some web components need additional patches to comply with UI5 framework requirements,
+	 * e.g. applying the LabelEnablement mixin
+	 *
+	 * Most things patched in this function should eventually be available generically in the custom elements metadta.
+	 *
+	 * @param {object} ui5metadata the UI5 metadata object
+	 */
+	#patchUI5Specifics(ui5metadata) {
+		const { tag } = ui5metadata;
+
+		// The label has a couple of specifics that are not fully reflected in the custom elements.
+		// This code harmonizes the Label with the original retrofit control wrapper.
+		// TODO: How to generalize this?
+		//       Is harmonizing with the retrofit library really correct?
+		if (tag === "ui5-label") {
+			// the ui5-label has as default slot, but no aggregations on the retrofit layer...
+			ui5metadata.aggregations = [];
```

Comment:
Everything except adding the interface in line :438, might not be needed if we decide to keep the naming as `for`.

Only difference is that `for` is a property, where as I think it should be an association in UI5.
Seems like a special case where the type is not `HTMLElement` (as it is with other associations). The `for` property is of type `string | undefined`. According to the documentation, the attribute value must be an ID.

I guess that should then be an association in UI5... 🤔

Reply:
So turns out we do need this harmonization for the Label control :(
The `sap/ui/core/LabelEnablement` expects an association named `labelFor`.

Reply:
Discussed and approved by PM.
Later down the line we can introduce a specific metadata setting.